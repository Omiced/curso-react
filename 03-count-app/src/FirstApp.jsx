import PropTypes from "prop-types";
//Se recomienda siempre destructurar las props
export const FirstApp = ({title,subtitle,nombre}) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{nombre}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  nombre: PropTypes.string
}

FirstApp.defaultProps = {
  // title : "no hay titulo"
  subtitle: "que onda",
  nombre: "perro"
}