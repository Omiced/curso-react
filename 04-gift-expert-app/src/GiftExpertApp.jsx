import { useState } from "react";
import {AddCategory, GiftGrid} from "./components";
import { RemoveCategory } from "./components/RemoveCategory";
export const GiftExpertApp = () => {
  
  const [categories, setCategories] = useState(['One Punch']);
  const onAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return;
    setCategories([newCategory,...categories]);
    
  }

  const onRemoveCategory =(removeCategory)=> {
    if(!removeCategory)return;
    const categoriesNew = categories.filter(categorie => categorie !== removeCategory);
    setCategories(categoriesNew);
  }

  return (
    <>
      <h1>Gift Expert App</h1>
      <AddCategory onNewCategory={onAddCategory}/>
      {categories.map(category => {
        return( 
        <>
         <GiftGrid key={category} category={category}/>   
         <RemoveCategory category={category}  key={category + "button"} onRemoveCategory={onRemoveCategory} />
        </>
      )
      })
      }
    </>
  );
};
