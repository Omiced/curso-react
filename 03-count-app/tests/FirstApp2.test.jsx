import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas firstApp', () => { 
    const title = "holo perrin";
    test('Debe hacer match con el snapshot', () => { 
      const {container} = render(<FirstApp title={title}/>);
      expect(container).toMatchSnapshot();
    });
    
    test('Debe de mostrar el mensaje "Holo perrin"', () => { 
      render(<FirstApp title={title}/>);
      //screen.debug() para ver el objeto screen
      expect(screen.getByText(title)).toBeTruthy();
     })

    test('Dede de mostrar el titulo en un h1', () => { 
      render(<FirstApp title={title}/>);
      expect(screen.getByRole('heading',{level:1}).textContent).toContain(title);
    })
 })