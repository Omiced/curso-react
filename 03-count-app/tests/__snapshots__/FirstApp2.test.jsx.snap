// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas firstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    holo perrin
  </h1>
  <p>
    que onda
  </p>
  <p>
    perro
  </p>
</div>
`;
